app/src/main/java/org/mozilla/fenix/home/sessioncontrol/viewholders/onboarding/OnboardingTrackingProtectionViewHolder.kt

```diff
@@ -7,17 +7,49 @@ package org.mozilla.fenix.home.sessioncontrol.viewholders.onboarding
 import android.view.View
 import androidx.recyclerview.widget.RecyclerView
 import kotlinx.android.synthetic.main.onboarding_tracking_protection.view.*
+import org.jetbrains.anko.dimen
 import org.mozilla.fenix.R
+import org.mozilla.fenix.ext.components
+import org.mozilla.fenix.utils.Settings
 
-class OnboardingTrackingProtectionViewHolder(view: View) : RecyclerView.ViewHolder(view) {
+class OnboardingTrackingProtectionViewHolder(val view: View) : RecyclerView.ViewHolder(view) {
 
     init {
+        val icon = view.context.getDrawable(R.drawable.ic_onboarding_tracking_protection)
+        val size = view.context.dimen(R.dimen.onboarding_header_icon_height_width)
+        icon?.setBounds(0, 0, size, size)
+
+        view.header_text.setCompoundDrawables(icon, null, null, null)
+
         val appName = view.context.getString(R.string.app_name)
         view.description_text.text = view.context.getString(
             R.string.onboarding_tracking_protection_description,
             appName
         )
+
+        val switch = view.tracking_protection_toggle
+
+        switch.isChecked = Settings.getInstance(view.context).shouldUseTrackingProtection
+
+        switch.setOnCheckedChangeListener { _, isChecked ->
+            updateTrackingProtectionSetting(isChecked)
+        }
     }
+
+    private fun updateTrackingProtectionSetting(enabled: Boolean) {
+        Settings.getInstance(view.context).setTrackingProtection(enabled)
+        with(view.context.components.core) {
+            val policy = createTrackingProtectionPolicy(enabled)
+            engine.settings.trackingProtectionPolicy = policy
+            sessionManager.sessions.forEach {
```

Comment:
There's a new use case for this so we don't have to go through each session see: https://github.com/mozilla-mobile/fenix/commit/cfca683a14ddc06bd09ea320bb5710cdcf001c80#diff-419111675b2c7ef0fa1219710d528e60R37